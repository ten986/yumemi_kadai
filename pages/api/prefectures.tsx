/* eslint-disable @typescript-eslint/explicit-module-boundary-types */
import axios from 'axios'
import { NextApiRequest, NextApiResponse } from 'next'

// APIの返り値の型
type PrefectureResponse = {
  message: string
  result: Prefecture[]
}

// APIの返り値の型
type Prefecture = {
  prefCode: number
  prefName: string
}

const handler = async (
  _req: NextApiRequest,
  res: NextApiResponse<Prefecture[]>,
) => {
  const result = await axios
    .get<PrefectureResponse>(
      'https://opendata.resas-portal.go.jp/api/v1/prefectures',
      {
        headers: {
          'X-API-KEY': process.env.X_API_KEY,
        },
      },
    )
    .then((value) => {
      return value.data.result
    })
  res.status(200).json(result)
}

export default handler
export type { Prefecture }
