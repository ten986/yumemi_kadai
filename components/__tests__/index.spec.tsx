import React from 'react'
import fetchMock from 'fetch-mock'
import { act } from 'react-dom/test-utils'
import { mount } from 'enzyme'

import IndexPage from 'pages/index'
import { Prefecture } from 'pages/api/prefectures'

const prefectures: Prefecture[] = [
  {
    prefCode: 1,
    prefName: '北海道',
  },
  {
    prefCode: 2,
    prefName: '青森県',
  },
  {
    prefCode: 3,
    prefName: '岩手県',
  },
  {
    prefCode: 4,
    prefName: '宮城県',
  },
]

describe('IndexPage', () => {
  beforeEach(() => {
    // mock の設定
    fetchMock.get(
      // リクエスト
      '/api/prefectures',
      // レスポンス
      {
        status: 200,
        body: prefectures,
      },
    )
  })
  afterEach(() => {
    // mock をリセット
    fetchMock.restore()
  })
  it('スナップショットが一致するか', async () => {
    const component = mount(<IndexPage />)
    // waitForComponentToPaint
    await act(async () => {
      await new Promise((resolve) => setTimeout(resolve, 0))
      component.update()
    })

    // スナップショットの一致
    expect(component).toMatchSnapshot()
  })
})
