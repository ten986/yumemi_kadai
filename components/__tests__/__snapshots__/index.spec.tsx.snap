// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`IndexPage スナップショットが一致するか 1`] = `
<IndexPage>
  <Header>
    <header
      className="header"
    >
      <div
        className="inner"
      >
        各都道府県の人口を表示するアプリ
      </div>
    </header>
  </Header>
  <div
    className="page"
  >
    <div>
      <h1>
        都道府県
      </h1>
      <CheckboxList
        checkedItems={Map {}}
        prefectures={
          Array [
            Object {
              "prefCode": 1,
              "prefName": "北海道",
            },
            Object {
              "prefCode": 2,
              "prefName": "青森県",
            },
            Object {
              "prefCode": 3,
              "prefName": "岩手県",
            },
            Object {
              "prefCode": 4,
              "prefName": "宮城県",
            },
          ]
        }
        setCheckedItems={[Function]}
      >
        <div
          className="wrapper"
        >
          <CheckBox
            checked={false}
            id="checkbox_1"
            key="0"
            onChange={[Function]}
            value="北海道"
          >
            <div
              className="wrapper"
            >
              <input
                checked={false}
                id="checkbox_1"
                name="checkbox"
                onChange={[Function]}
                type="checkbox"
                value="北海道"
              />
              <label
                htmlFor="checkbox_1"
              >
                北海道
              </label>
            </div>
          </CheckBox>
          <CheckBox
            checked={false}
            id="checkbox_2"
            key="1"
            onChange={[Function]}
            value="青森県"
          >
            <div
              className="wrapper"
            >
              <input
                checked={false}
                id="checkbox_2"
                name="checkbox"
                onChange={[Function]}
                type="checkbox"
                value="青森県"
              />
              <label
                htmlFor="checkbox_2"
              >
                青森県
              </label>
            </div>
          </CheckBox>
          <CheckBox
            checked={false}
            id="checkbox_3"
            key="2"
            onChange={[Function]}
            value="岩手県"
          >
            <div
              className="wrapper"
            >
              <input
                checked={false}
                id="checkbox_3"
                name="checkbox"
                onChange={[Function]}
                type="checkbox"
                value="岩手県"
              />
              <label
                htmlFor="checkbox_3"
              >
                岩手県
              </label>
            </div>
          </CheckBox>
          <CheckBox
            checked={false}
            id="checkbox_4"
            key="3"
            onChange={[Function]}
            value="宮城県"
          >
            <div
              className="wrapper"
            >
              <input
                checked={false}
                id="checkbox_4"
                name="checkbox"
                onChange={[Function]}
                type="checkbox"
                value="宮城県"
              />
              <label
                htmlFor="checkbox_4"
              >
                宮城県
              </label>
            </div>
          </CheckBox>
        </div>
      </CheckboxList>
    </div>
  </div>
</IndexPage>
`;
